### ISSUE TITLE
Losing Audio and Video in Live conf when "Start recording"

### ISSUE THREAD

**Issue by iasbm** [jitsi/docker-jitsi-meet#1550]
Hello,
Before we go into the actual question i better explain our current setup and env so that its clear.
```
ENV -->
JITSI HELM Charts along with JIBRI, JICOFO, PROSODY,WEB, JVB and JIGASI installed in K8S cluster.
```
We implemented React SDK based JITSI Conference set up as explained [here ](https://jitsi.github.io/handbook/docs/dev-guide/dev-guide-react-sdk)

While being in Conference, All ( calls, chats, screen sharing) is happening as expected.   But when we "start recording" then we are losing audio and video in live conference itself.  Initially we thought it might be some issue with our Coding or COnfig.

So we went to our custom domain hosted JITSI url ( like jistsi.mycustomdomain.com) and start Conference there so that we can eliminate the possible coding, config errors in our React Code.  And there also same behaviour and the moment we start recording both participants are losing their video and audio.  Then once we stop recording then audio and video are back.

thanks in advance for any insights on this pls.

**Comment by damencho** [comment]
Are you testing with 2 participants and starting a recording? 
Try testing just with 3 participants, no recording, do you still see an issue?

**Comment by iasbm** [comment]
Yes we can try our custom domain hosted JITSI url ( like jistsi.mycustomdomain.com) with 3 participants but we still need Recording.
But our actual business use case only requires 2 users and Recording is a must for 2 participatns also.

so right now all is working including audio, video if we don't do "Start Recording" but once we start recording the participants lose Audio, Video and they will get them back only when we stop recording.  This happens on our custom application (where we implemented React based hosting ) and also on our custom domain hosted JITSI url ( like jistsi.mycustomdomain.com) too

**Comment by damencho** [comment]
> we just tested with 3 participants on our custom domain hosted JITSI url ( like jistsi.mycustomdomain.com) , but we can't see video of other 2 participants and no Audio even witout START RECORDIGN.

Oh, yeah. This proves you have problem and udp on port 10000 is not reaching the jvb. 
It is either jvb advertising wrong IP or you have a firewall or missing port forward on the device handling the public IP.

**Comment by iasbm** [comment]
Ok just executed the following scenario....
Step-1.  Started Conference with 2 participants Person1 and Person-2.
Step-2  Both can see each other. 
Step-3 Now Person-1 started Recording both participants lost Videos.
Step-4  Person-1 stopped recording. 

Final result for couple of seconds conference is back to normal but Person-2 lost video of person-1

**Comment by damencho** [comment]
This proves you have problem and udp on port 10000 is not reaching the jvb when media is sent from clients.
It is either jvb advertising wrong IP or you have a firewall or missing port forward on the device handling the public IP.

I have no idea about Helm Charts but the above should help you find the problem. 

Make sure your jvb is running with correct settings for this:
https://github.com/jitsi/docker-jitsi-meet/blob/e4c8c2a963f0267ff07a6e2a2df038d456f8e171/env.example#L37
This will instruct jvb to advertise the correct public ip address to clients, and that public address needs to forward udp port 10000 to jvb.

https://github.com/jitsi/docker-jitsi-meet/blob/e4c8c2a963f0267ff07a6e2a2df038d456f8e171/jvb/rootfs/defaults/jvb.conf#L119

Although by default you should have this https://github.com/jitsi/docker-jitsi-meet/blob/e4c8c2a963f0267ff07a6e2a2df038d456f8e171/jvb/rootfs/defaults/jvb.conf#L113 which will allow jvb to find its public address automatically.

**Comment by iasbm** [comment]
whole of yesterday i have checked multiple times and i can confirm that our JVB node is getting its env var "JVB_ADVERTISE_IPS : fieldRef(v1:status.hostIP)" populated like this through HELM Charts.
In fact i have also tried update "publicIPs" property with public IP of JVB Node too, but no luck so far.

no matter what we try so far the result is same and participants are losing video and audio the moment we start recording and and once i stop recording the media is restored back to normal and this happens to all the participants in the Room.

Any other ideas or help is much appreciated pls.

**Comment by saghul** [comment]
I read through this and it gets confusing. Are you able to have a conference with 3 participants which has both audio and video and it doesn't stop?

**Comment by iasbm** [comment]
To be sure i have just now tried again with 3 participants in 3 separate incognito browsers.  And we can't have conference as there is NO media (both audio and video) for participants in the conference.   We didn't even attempt REcording and the conference just doesn't have Media at all.

So this is making me think in 2 participants mode we are able to have conference in P2P mode ( but again media lost when we attempt Recording) without JVB playing the role of SFU?  

But when we have 3 participants then the role of JVB as SFU is kicking in and as a result is there is NO media at all in conference even without attempt of Recording?

But as mentioned in my above message [here ](https://github.com/jitsi/docker-jitsi-meet/issues/1550#issuecomment-1552456040), we configured all Firewall ports of `UDP 10000` for all the nodes in the cluster and also i can confirm that our JVB node is getting its env var "`JVB_ADVERTISE_IPS : fieldRef(v1:status.hostIP)`" populated like this through HELM Charts.

Not sure why and what else are we missing here pls

**Comment by damencho** [comment]
Hummm someone recently mentioned something, was ot for Helm that it would change and the source address of the packages, not only the destination like a NAT and that break the jvb connectivity as it cannot send packets back to the clients. You may check that.

**Comment by iasbm** [comment]
thanks for some insights, any possible relevant links for this pls

**Comment by saghul** [comment]
Sorry we don't use k8s / helm ourselves, I don't know what to suggest.

**Comment by iasbm** [comment]
Just to keep you posted, we are still investigating the issue . But just wanted to attached the error we grabbed in screen shot if it provides any insight and opens any further thoughts pls

![JVB issues](https://github.com/jitsi/docker-jitsi-meet/assets/26871176/21a1e2f5-529a-4406-8d3f-1993a2050039)

**Comment by damencho** [comment]
You need to deploy a turn server to be able for the clients to send media over TCP which then will reach JVB over UDP. But mind that using TCP for media is normally used only as last resort/fallback when UDP is not possible, thia is because TCP is slow which will for sure degarde quality.

**Comment by iasbm** [comment]
ok thanks @damencho  for a prompt response,  let me go  back to my cloud provider if they can release UDP on request basis so that i can keep this TCP as a last option.  Appreciate your Suggestion.

**Comment by saghul** [comment]
I'm closing this since it's not a Jitsi issue but an issue in your provider.
